UnusedImport error if there are only array references to a given class

```bash
slovdahl@desk:~/checkstyle-unused-import-bug-reproducer/src/main/java (main)$ javac checkstyle/reproducer/UnusedImportBug.java 
slovdahl@desk:~/checkstyle-unused-import-bug-reproducer/src/main/java (main)$ 

slovdahl@desk:~/checkstyle-unused-import-bug-reproducer (main)$ cat gradle/checkstyle/checkstyle.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="UnusedImports">
            <property name="processJavadoc" value="true"/>
        </module>
    </module>
</module>


slovdahl@desk:~/checkstyle-unused-import-bug-reproducer (main)$ cat src/main/java/checkstyle/reproducer/UnusedImportBug.java 
package checkstyle.reproducer;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;

public class UnusedImportBug {

    private static final Set<String> FOO;
    static {
        FOO = new HashSet<>();

        FOO.add( HashMap[].class.getName() );
    }
}


slovdahl@desk:~/checkstyle-unused-import-bug-reproducer (main)$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
slovdahl@desk:~/checkstyle-unused-import-bug-reproducer (main)$ java $RUN_LOCALE -jar checkstyle-9.0.1-all.jar -c gradle/checkstyle/checkstyle.xml src/main/java/checkstyle/reproducer/UnusedImportBug.java
Starting audit...
[ERROR] /home/slovdahl/checkstyle-unused-import-bug-reproducer/src/main/java/checkstyle/reproducer/UnusedImportBug.java:3:8: Unused import - java.util.HashMap. [UnusedImports]
Audit done.
Checkstyle ends with 1 errors.

```

---

**Describe what you expect in detail.**

The `UnusedImport` error should not be triggered if an import is actually used.

---

MCVE: https://github.com/hiboxsystems/checkstyle-unused-import-bug-reproducer

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java
@@ -272,17 +272,13 @@ public class UnusedImportsCheck extends AbstractCheck {
                 || parentType == TokenTypes.METHOD_DEF;
 
         final boolean isQualifiedIdent = parentType == TokenTypes.DOT
+                && !TokenUtil.isOfType(ast.getPreviousSibling(), TokenTypes.DOT)
                 && ast.getNextSibling() != null;
-
-        final boolean isQualifiedNameArrayType = parent.getParent().getType() == TokenTypes.DOT
-                && ast.getNextSibling() != null
-                && ast.getNextSibling().getType() == TokenTypes.ARRAY_DECLARATOR;
 
         if (TokenUtil.isTypeDeclaration(parentType)) {
             currentFrame.addDeclaredType(ast.getText());
         }
-        else if ((!isPossibleDotClassOrInMethod || isQualifiedIdent)
-                    && !isQualifiedNameArrayType) {
+        else if (!isPossibleDotClassOrInMethod || isQualifiedIdent) {
             currentFrame.addReferencedType(ast.getText());
         }
     }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheckTest.java
@@ -146,6 +146,15 @@ public class UnusedImportsCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testArrayRef() throws Exception {
+        final String[] expected = {
+            "13:8: " + getCheckMessage(MSG_KEY, "java.util.ArrayList"),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputUnusedImportsArrayRef.java"), expected);
+    }
+
+    @Test
     public void testBug() throws Exception {
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
         verifyWithInlineConfigParser(

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/unusedimports/InputUnusedImportsArrayRef.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/unusedimports/InputUnusedImportsArrayRef.java
@@ -1,0 +1,24 @@
+/*
+UnusedImports
+processJavadoc = (default)true
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.imports.unusedimports;
+
+import java.util.HashMap; // ok
+import java.util.HashSet;
+import java.util.Set;
+import java.util.ArrayList; // violation
+
+public class InputUnusedImportsArrayRef {
+
+    private static final Set<String> FOO;
+    static {
+        FOO = new HashSet<>();
+
+        FOO.add( HashMap[].class.getName() );
+        FOO.add( java.util.ArrayList[].class.getName() );
+    }
+}

